libkbfs/mdserver_remote.go

```diff
@@ -171,34 +171,41 @@ func (md *MDServerRemote) resetAuth(ctx context.Context, c keybase1.MetadataClie
 	}()
 
 	session, err := md.config.KBPKI().GetCurrentSession(ctx)
-	if err != nil {
-		md.log.Debug("MDServerRemote: User logged out, skipping resetAuth")
-		return MdServerDefaultPingIntervalSeconds, NoCurrentSessionError{}
+	if _, ok := err.(NoCurrentSessionError); ok {
+		md.log.CDebugf(ctx, "User logged out (%+v), skipping resetAuth", err)
+		return MdServerDefaultPingIntervalSeconds, err
+	} else if err != nil {
+		// We can't rely on getting a `LoggedIn` error from the
+		// service in this case, so let the RPC connection fail and
+		// retry a bit later.
+		md.log.CDebugf(ctx, "Unexpected error when getting session: %+v", err)
+		return MdServerDefaultPingIntervalSeconds, err
```

Comment:
That wouldn't be bad if we were returning something different in the two cases, but it's confusing if we're just sending back the same error no matter what.